core/src/OpenACC/Kokkos_OpenACC_ParallelFor_MDRange.hpp

```diff
@@ -0,0 +1,338 @@
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_OPENACC_PARALLEL_FOR_MDRANGE_HPP
+#define KOKKOS_OPENACC_PARALLEL_FOR_MDRANGE_HPP
+
+#include <OpenACC/Kokkos_OpenACC.hpp>
+#include <OpenACC/Kokkos_OpenACC_FunctorAdapter.hpp>
+#include <Kokkos_Parallel.hpp>
+
+namespace Kokkos::Experimental::Impl {
+
+template <class Functor, class Policy, int Rank = Policy::rank>
+struct OpenACCParallelForHelper {
+  OpenACCParallelForHelper(Functor const&, Policy const&, int) {
+    static_assert(std::is_void_v<Functor>, "not implemented");
+  }
+};
+
+}  // namespace Kokkos::Experimental::Impl
+
+template <class Functor, class... Traits>
+class Kokkos::Impl::ParallelFor<Functor, Kokkos::MDRangePolicy<Traits...>,
+                                Kokkos::Experimental::OpenACC> {
+  using Policy = MDRangePolicy<Traits...>;
+  Kokkos::Experimental::Impl::FunctorAdapter<Functor, Policy> m_functor;
+  Policy m_policy;
+
+ public:
+  ParallelFor(Functor const& functor, Policy const& policy)
+      : m_functor(functor), m_policy(policy) {}
+
+  void execute() const {
+    static_assert(Policy::rank < 7 && Policy::rank > 1,
+                  "OpenACC Backend MDRangePolicy Error: Unsupported rank...");
+    static_assert(Policy::inner_direction == Iterate::Left ||
+                  Policy::inner_direction == Iterate::Right);
+    int const async_arg = m_policy.space().acc_async_queue();
+    Kokkos::Experimental::Impl::OpenACCParallelForHelper(m_functor, m_policy,
+                                                         async_arg);
+  }
+};
+
+template <class Functor, class... Traits>
+struct Kokkos::Experimental::Impl::OpenACCParallelForHelper<
+    Functor, Kokkos::MDRangePolicy<Traits...>, 2> {
+  using Policy = MDRangePolicy<Traits...>;
+  OpenACCParallelForHelper(Functor const& functor, Policy const& policy,
+                           int async_arg) {
+    auto const begin1 = policy.m_lower[1];
+    auto const end1   = policy.m_upper[1];
+    auto const begin0 = policy.m_lower[0];
+    auto const end0   = policy.m_upper[0];
+
+    if ((end0 <= begin0) || (end1 <= begin1)) {
+      return;
+    }
+
+    if constexpr (Policy::inner_direction == Iterate::Left) {
+#pragma acc parallel loop gang vector collapse(2) copyin(functor) \
+    async(async_arg)
+      for (auto i1 = begin1; i1 < end1; ++i1) {
+        for (auto i0 = begin0; i0 < end0; ++i0) {
+          functor(i0, i1);
+        }
+      }
+    } else if constexpr (Policy::inner_direction == Iterate::Right) {
+#pragma acc parallel loop gang vector collapse(2) copyin(functor) \
+    async(async_arg)
+      for (auto i0 = begin0; i0 < end0; ++i0) {
+        for (auto i1 = begin1; i1 < end1; ++i1) {
+          functor(i0, i1);
+        }
+      }
+    } else {
+      static_assert(
+          "Kokkos Error: not supported memory layout for the OpenACC backend");
```

Comment:
Did you mean
```suggestion
      static_assert(false,
          "Kokkos Error: not supported memory layout for the OpenACC backend");
```
? Any string implicitly converts to `true`.

Reply:
Thank you for catching this; updated.

Reply:
Changed `static_assert(false, …)` to `Kokkos::abort(…)` since `static_assert(false, …)` will always complain errors.